src/applications/personalization/preferences/actions/index.js

```diff
@@ -1,33 +1,138 @@
-import localStorage from 'platform/utilities/storage/localStorage';
+import { apiRequest } from 'platform/utilities/api';
 
-export const DASHBOARD_PREFERENCE_SET = 'DASHBOARD_PREFERENCE_SET';
-export const DASHBOARD_PREFERENCES_SAVED = 'DASHBOARD_PREFERENCES_SAVED';
-export const DASHBOARD_PREFERENCES_FETCHED = 'DASHBOARD_PREFERENCES_FETCHED';
+import { LOADING_STATES, PREFERENCE_CODES } from '../constants';
+import { transformPreferencesForSaving } from '../helpers';
 
-export function fetchPreferences() {
-  const savedPrefs = localStorage.getItem('dashboard-preferences');
-  localStorage.setItem('dashboardLastVisitedAt', Date.now());
+export const SET_USER_PREFERENCE_REQUEST_STATUS =
+  'SET_USER_PREFERENCE_REQUEST_STATUS';
+export const SET_ALL_PREFERENCE_OPTIONS_REQUEST_STATUS =
+  'SET_ALL_PREFERENCE_OPTIONS_REQUEST_STATUS';
+export const SET_SAVE_PREFERENCES_REQUEST_STATUS =
+  'SET_SAVE_PREFERENCES_REQUEST_STATUS';
+export const SET_AVAILABLE_BENEFITS = 'SET_AVAILABLE_BENEFITS';
+export const SET_DASHBOARD_USER_PREFERENCES = 'SET_DASHBOARD_USER_PREFERENCES';
+export const SET_DASHBOARD_PREFERENCE = 'SET_DASHBOARD_PREFERENCE';
+export const SAVED_DASHBOARD_PREFERENCES = 'SAVED_DASHBOARD_PREFERENCES';
 
-  return {
-    type: DASHBOARD_PREFERENCES_FETCHED,
-    data: (savedPrefs && JSON.parse(savedPrefs)) || {},
+// load the benefits the user has picked to learn more about
+export function fetchUserSelectedBenefits() {
+  return dispatch => {
+    dispatch({
+      type: SET_USER_PREFERENCE_REQUEST_STATUS,
```

Comment:
Idiomatic redux forks the behavior with different action types i.e. it uses on variable on the action instead of many. Couple of issues with this: 
- the tooling is built around actions- if you use redux dev tools with this you will see a log of the same action twice and will need to dig into it to see if the app is behaving the way you expect
- I don't see any reducers for this action but I assume this will fork the behavior with a switch and an if vs reducers usually forking with a single switch statement. 

We have an open issue from @jbalboni about this: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/10845

Reply:
That action type is handled by the first reducer. But I like the approach you are advocating of having a single action type for each event.